### ISSUE TITLE
Cargo EjectOnDeath not always working

### ISSUE THREAD

**Issue by oldnaari** [OpenRA/OpenRA#20780]
I am working on a mod, which has a custom transport unit. Below is its .yaml.
For it the `EjectOnDeath` seems not always work.

```
shuttle:
	Inherits: ^Tank
	Inherits@GAINSEXPERIENCE: ^GainsExperience
	Buildable:
		Queue: Vehicle
		BuildPaletteOrder: 20
		Prerequisites: ~light.smuggler, upgrade.light, high_tech_factory, ~techlevel.medium
		BuildDuration: 225
		BuildDurationModifier: 100
		Description: Transport unit
	Encyclopedia:
		Description: Shuttles are transport units, disguised under Arakeen transport. When the shuttle is destroyed it drops the infantry units it contains.\n\nIt can run over infantry units, despite being produced from the light factory
		Order: 125
		Category: Units
	Valued:
		Cost: 300
	Tooltip:
		Name: Shuttle
	UpdatesPlayerStatistics:
		AddToArmyValue: true
	Selectable:
		Class: shuttle
	Health:
		HP: 3000
	Armor:
		Type: wood
	Mobile:
		TurnSpeed: 28
		Speed: 80
		Locomotor: tank
		PauseOnCondition: notmobile
	Cargo:
		Types: Infantry
		MaxWeight: 3
		LoadingCondition: notmobile
		EjectOnDeath: true
	RevealsShroud:
		Range: 4c768
	Explodes:
		Weapon: UnitExplodeSmall
		EmptyWeapon: UnitExplodeSmall
	AttractsWorms:
		Intensity: 420
	WithCargoPipsDecoration:
		Position: BottomLeft
		Margin: 1, 4
		RequiresSelection: true
		PipCount: 3
````

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/31252576/229301622-954d73aa-0c42-493a-b35a-f7b103cb0c5d.gif)

**Comment by PunkPun** [comment]
duplicate of #16353
fixed by #20515
